flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/expressions/RexNodeConverter.java

```diff
@@ -168,6 +170,12 @@ private RexNode visitScalarFunc(UnresolvedCallExpression unresolvedCall) {
 			return relBuilder.call(FlinkSqlOperatorTable.MULTIPLY, child);
 		} else if (BuiltInFunctionDefinitions.MOD.equals(def)) {
 			return relBuilder.call(FlinkSqlOperatorTable.MOD, child);
+		} else if (def instanceof ScalarFunctionDefinition) {
+			// TODO add getName for ScalarFunctionDefinition ???
```

Comment:
It looks wrong code, should be:
```
ScalarFunction scalaFunc = ((ScalarFunctionDefinition) func).getScalarFunction();
			List<RexNode> child = convertCallChildren(call);
			SqlFunction sqlFunction = UserDefinedFunctionUtils.createScalarSqlFunction(
					scalaFunc.functionIdentifier(),
					scalaFunc.toString(),
					scalaFunc,
					typeFactory);
			return relBuilder.call(sqlFunction, child);
```

Reply:
thanks for the correction. 
`List<RexNode> child = convertCallChildren(call);` is defined at line 122

Reply:
Now we just use `functionIdentifier` to be name.
The detail reason is in https://github.com/apache/flink/pull/3330

Reply:
we can use `functionIdentifier` now, and fix it in another issue